phoenix-core/src/main/java/org/apache/phoenix/execute/MutationState.java

```diff
@@ -1572,12 +1572,19 @@ public static MutationMetricQueue.MutationMetric updateMutationBatchFailureMetri
                 numDeleteMutationsInBatch++;
             }
         }
+
+        long totalfailedMutation = numUpsertMutationsInBatch + numDeleteMutationsInBatch;
+        //TODO this case is something that should not happen
+        // but still if connditioon makes sese if this ever happens
+        if (totalfailedMutation < numFailedMutations) {
+            totalfailedMutation = numFailedMutations;
+        }
```

Comment:
As per this comment, we will likely not hit the case where `totalfailedMutation < numFailedMutations`?

Reply:
That is right. That is the understanding.

Reply:
Added warning log there.